python/ray/tests/test_actor_lineage_reconstruction.py

```diff
@@ -102,6 +104,60 @@ def verify3():
     wait_for_condition(lambda: verify3())
 
 
+def test_mix_actor_restart_task_retry_and_lineage_reconstruction(ray_start_cluster):
+    # This test simulates a scenario where the environment aggressively
+    # preempts the worker node.
+    #
+    # It creates a cluster with 1 head and 1 worker node. Then, it deploys
+    # an actor `TestActor` on the worker with infinite actor restarts and
+    # task retries. The actor will only be scheduled on the worker node.
+    #
+    # Next, it repeatedly kills the worker node to trigger lineage
+    # reconstruction, actor restarts, and task retries. This test ensures
+    # that these three mechanisms can work together.
+    cluster = ray_start_cluster
+    cluster.add_node(resources={"head": 1})
+    ray.init(address=cluster.address)
+
+    worker_node = cluster.add_node(resources={"worker": 1})
+
+    class BigObject:
+        def __init__(self, idx):
+            self.idx = idx
+            self.data = b"x" * 1000 * 200
+
+    @ray.remote(resources={"worker": 1}, max_restarts=-1, max_task_retries=-1)
+    class TestActor:
+        def __init__(self):
+            self.done = set()
+
+        def f(self, idx):
+            # Make sure each task is executed only once on a single actor
+            # instance
+            assert idx not in self.done
+            self.done.add(idx)
+            time.sleep(1)
+            return BigObject(idx)
+
+    actor = TestActor.remote()
+
+    refs = []
+    for i in range(50):
+        refs.append(actor.f.remote(i))
+
+    iter = 10
+    for i in range(iter):
+        time.sleep(2)
```

Comment:
please try to write the test in a way that doesn't rely on `time.sleep` here or inside of `f`. if you need help with how to do this, lmk